src/api/db/migrate/20180405074538_change_default_charset_and_mysql57_defaults.rb

```diff
@@ -0,0 +1,45 @@
+class ChangeDefaultCharsetAndMysql57Defaults < ActiveRecord::Migration[5.1]
+  # mysql/mariadb has changed the default to utf8mb4, we need to update old instances
+  # to avoid to have specific sql code for difference mysql versions
+  def up
+    # these may be utf8 or utf8mb4 depending on the use mysql instance on creation time
+    [ 'ar_internal_metadata', 'binary_releases', 'bs_request_counter',
+      'cloud_azure_configurations', 'cloud_ec2_configurations', 
+      'cloud_user_upload_jobs', 'data_migrations', 'download_repositories',
+      'group_maintainers', 'incident_updateinfo_counter_values',
+      'kiwi_images', 'kiwi_package_groups', 'kiwi_packages',
+      'kiwi_preferences', 'kiwi_repositories', 'maintained_projects',
+      'product_media', 'product_update_repositories' ].each do |table|
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8");
+    end
+    # contains free text, could have emojis
```

Comment:
*emojis* can mean a lot. There are emojis inside utf8 and emojis outside utf8. What are you actually trying to solve?

Inside:
```ruby
Configuration.first.update_attributes(description: '☭')
  Configuration Load (0.5ms)  SELECT  `configurations`.* FROM `configurations` ORDER BY `configurations`.`id` ASC LIMIT 1
   (0.1ms)  BEGIN
  SQL (0.3ms)  INSERT INTO `delayed_jobs` (`handler`, `run_at`, `queue`) VALUES ('--- !ruby/object:ActiveJob::QueueAdapters::DelayedJobAdapter::JobWrapper\njob_data:\n  job_class: ConfigurationWriteToBackendJob\n  job_id: 2d52a64b-51dd-473f-9e98-31cf9b2da15b\n  provider_job_id: \n  queue_name: quick\n  priority: \n  arguments:\n  - 1\n  executions: 0\n  locale: en\n', '2018-04-09 09:27:59', 'quick')
Enqueued ConfigurationWriteToBackendJob (Job ID: 2d52a64b-51dd-473f-9e98-31cf9b2da15b) to DelayedJob(quick) with arguments: 1
   (3.5ms)  COMMIT
=> true
```

Outside:

```ruby
Configuration.first.update_attributes(description: '🐝🐜🐛')
   (0.3ms)  SET NAMES utf8,  @@SESSION.sql_mode = CONCAT(CONCAT(@@sql_mode, ',STRICT_ALL_TABLES'), ',NO_AUTO_VALUE_ON_ZERO'),  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483
  Configuration Load (0.3ms)  SELECT  `configurations`.* FROM `configurations` ORDER BY `configurations`.`id` ASC LIMIT 1
   (0.3ms)  BEGIN
  SQL (0.8ms)  UPDATE `configurations` SET `description` = '🐝🐜🐛', `updated_at` = '2018-04-09 09:23:35.997072' WHERE `configurations`.`id` = 1
   (0.3ms)  ROLLBACK
Traceback (most recent call last):
        1: from (irb):1
ActiveRecord::StatementInvalid (Mysql2::Error: Incorrect string value: '\xF0\x9F\x90\x9D\xF0\x9F...' for column 'description' at row 1: UPDATE `configurations` SET `description` = '🐝🐜🐛', `updated_at` = '2018-04-09 09:23:35.997072' WHERE `configurations`.`id` = 1)
```

Reply:
was only an example, it was intended to cover all tables where utf8mb4 might be needed.

Reply:
> it was intended to cover all tables where utf8mb4 might be needed.

Hm, I'm not a 100% sure how you came to the conclusion that those are the tables. Can you explain?